src/compiler/emitter.ts

```diff
@@ -2492,7 +2496,75 @@ module ts {
                 write("]");
             }
 
+            function hasSpreadElement(elements: Expression[]) {
+                return forEach(elements, e => e.kind === SyntaxKind.SpreadElementExpression);
+            }
+
+            function skipParentheses(node: Expression): Expression {
+                while (node.kind === SyntaxKind.ParenthesizedExpression || node.kind === SyntaxKind.TypeAssertionExpression) {
+                    node = (<ParenthesizedExpression>node).expression;

```

Comment:
This type assertion should be to `ParenthesizedExpression | TypeAssertionExpression`.